### DISCUSSION TITLE
Mongoid extra

### DISCUSSION THREAD

**Discussion by merouaneamqor** [ddnexus/pagy#670] [Feature Requests]
### Description

```ruby
# frozen_string_literal: true

module BackendPagyExtension
  private
    def pagy(collection, vars = {})
      pagy = Pagy.new(pagy_get_vars(collection, vars))
      [pagy, pagy_get_items(collection, pagy)]
    end

    def pagy_get_vars(collection, vars)
      pagy_set_items_from_params(vars) if defined?(Pagy::Extras::Items)

      vars[:count] ||= optimized_count_for(collection)

      vars[:page] ||= params[vars[:page_param] || Pagy::DEFAULT[:page_param]]
      vars
    end

    def pagy_get_items(collection, pagy)
      collection.offset(pagy.offset).limit(pagy.items)
    end

    def optimized_count_for(collection)
      if collection.respond_to?(:count_documents)
        collection.count_documents
      else
        count = collection.count
        count.is_a?(Hash) ? count.size : count
      end
    end
end
```

**Comment by ddnexus** [comment]
After [8.2.0](https://github.com/ddnexus/pagy/releases/tag/8.2.0), all the code above could be reduced to a simple overriding of the `pagy_get_count` new method in your application controller:

```ruby
def pagy_get_count(collection, vars)
  collection.respond_to?(:count_documents) ? collection.count_documents : super
end
```

@benkoshy We could add the example above somewhere in the doc. Maybe we can add it in the `/docs/api/pagy`, and/or in the `/docs/how-to` something like "Using count_documents with Mongoid".
 
After we do that, we can close this discussion. 

@merouaneamqor?

**Reply by benkoshy** [reply]
> Maybe we can add it in the /docs/api/pagy, and/or in the /docs/how-to something like "Using count_documents with Mongoid".

My concern is discoverability: how will mongo users find it?

If it's not an extra with code to back it, but purely a documentation concern, any objections to throwing it in as a faux-extra?

![image](https://github.com/ddnexus/pagy/assets/15097447/1aef66ea-fe57-498a-ab2d-8db3c29b7f49)

OR another alternative is to throw it in here: pagy override instructions?

![image](https://github.com/ddnexus/pagy/assets/15097447/2cfc1b5e-e930-47ae-96f5-ed59fad8cd5a)

**Comment by ddnexus** [comment]
Discoverability should not be a concern because you just search for "mongoid" in the doc. I don't think anybody of sound mind would "browse" instead of searching for a specific narrow thing.

We already have too long menus in the doc, particularly in the extras.

**Reply by benkoshy** [reply]
#691
